src/segment-loader.js

```diff
@@ -246,7 +216,9 @@ export default class SegmentLoader extends videojs.EventTarget {
     this.syncController_.on('syncinfoupdate', () => this.trigger('syncinfoupdate'));
 
     this.mediaSource_.addEventListener('sourceopen', () => {
-      this.ended_ = false;
+      if (!this.isEndOfStream_()) {
```

Comment:
This prevents `ended_` from being set to false when this segment loader is still ended, but another opens the mediaSource to append more data. IE this can happen when switching audio track after end of stream.